Scripts/datatypes/purpose.py

```diff
@@ -296,6 +295,7 @@ def calc_prob(self, impedance, is_last_iteration):
             self.accessibility_model.calc_accessibility(
                 copy(purpose_impedance))
         self.prob = self.model.calc_prob(purpose_impedance)
+        log.info(f"Mode and dest probabilities calculated for {self.name}")
```

Comment:
This is part of the first heavy loop of the demand model and `calc_prob` is the most time-consuming step in that loop, so it makes sense to log when that step is done.